src/common/telemetry.rs

```diff
@@ -352,25 +472,84 @@ impl TelemetryCollector {
     fn get_web_data(&self) -> WebApiTelemetry {
         let actix_telemetry_collector = self.actix_telemetry_collector.lock();
         let mut result = WebApiTelemetry::default();
-        for web_data in &actix_telemetry_collector.web_workers_telemetry {
-            let lock = web_data.lock();
+        for web_data in &actix_telemetry_collector.workers {
+            let lock = web_data.lock().get_telemetry_data();
             result.merge(&lock);
         }
         result
     }
 
-    fn grpc_calls_statistics(&self) -> TelemetryOperationStatistics {
+    fn grpc_calls_statistics(&self) -> GrpcTelemetry {
         let tonic_telemetry_collector = self.tonic_telemetry_collector.lock();
-        let mut result = TelemetryOperationStatistics::default();
-        for grps_calls in &tonic_telemetry_collector.grpc_calls_aggregators {
-            let stats = grps_calls.lock().get_statistics();
-            result.ok_count += stats.ok_count;
-            result.fail_count += stats.fail_count;
-            result.ok_avg_time += stats.ok_avg_time;
-        }
-        if !tonic_telemetry_collector.grpc_calls_aggregators.is_empty() {
-            result.ok_avg_time /= tonic_telemetry_collector.grpc_calls_aggregators.len() as u32;
+        let mut result = GrpcTelemetry::default();
+        for grpc_data in &tonic_telemetry_collector.workers {
+            let lock = grpc_data.lock().get_telemetry_data();
+            result.merge(&lock);
         }
         result
     }
 }
+
+impl TelemetryData {
+    pub fn agregate(&mut self) {
+        let mut collections_short_info = CollectionShortInfoTelemetry::default();
+        for collection in self.collections.as_mut().unwrap().iter_mut() {
```

Comment:
Why is this unwrap safe?

Reply:
I removed unwrap usages in this function. Unwrap was safe but I believe that non-obvious "save unwraps" are bad too